src/Icicle/Pipeline.hs

```diff
@@ -81,7 +93,7 @@ annotOfError e
     CompileErrorProgram _
      -> Nothing
 
-instance Pretty a => Pretty (CompileError a) where
+instance (IsString b, Ord b, Pretty a, Pretty b, Show a, Show b, Show c) => Pretty (CompileError a b c) where
```

Comment:
the ``IsString`` and ``Ord`` seem strange to me. are they really necessary? I can't think offhand why pretty printing an error would require making a variable, so if it is necessary, add a comment please